### PR TITLE
Document how to use QuarkusTestProfile#testResources

### LINKED ISSUE #12081: Document how to use QuarkusTestProfile.testResources()
**Description**
This cool and handy feature was introduced via #10368 and it deserves to be documented.

In fact today a colleague of mine was irritated and tried to use both `@TestProfile` _and_ `@QuarkusTestResource` for his test class (which doesn't work).

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #12081

### CODE REVIEW

**Comment by geoand** [comment]
cc @famod

File: docs/src/main/asciidoc/getting-started-testing.adoc [hunk 1]
```diff
@@ -504,11 +505,19 @@ public class MockGreetingProfile implements QuarkusTestProfile {
     public String getConfigProfile() { <3>
         return "test";
     }
+
+    @Override
+    public List<TestResourceEntry> testResources() { <4>
+        return Collections.emptyList();
```
Comment (famod) [inline docs/src/main/asciidoc/getting-started-testing.adoc:511]: Maybe something like `return List.of(new TestResourceEntry(MyWireMockServerManager.class));` or similar would make the usage even more clear.

**Review by famod — APPROVED** [review]

File: docs/src/main/asciidoc/getting-started-testing.adoc (hunk 1 above)
Comment (geoand) [reply docs/src/main/asciidoc/getting-started-testing.adoc]: Good idea!

File: docs/src/main/asciidoc/getting-started-testing.adoc [hunk 2]
```diff
@@ -504,11 +505,19 @@ public class MockGreetingProfile implements QuarkusTestProfile {
     public String getConfigProfile() { <3>
         return "test";
     }
+
+    @Override
+    public List<TestResourceEntry> testResources() { <4>
+        return Collections.SingletonList(new TestResourceEntry(CustomWireMockServerManager.class));
```
Comment (gastaldi) [inline docs/src/main/asciidoc/getting-started-testing.adoc:511]:
```suggestion
        return Collections.singletonList(new TestResourceEntry(CustomWireMockServerManager.class));
```

Comment (geoand) [reply docs/src/main/asciidoc/getting-started-testing.adoc]: Thanks for spotting that!

**Review by gastaldi — APPROVED** [review]

File: docs/src/main/asciidoc/getting-started-testing.adoc [hunk 3]
```diff
@@ -504,11 +505,19 @@ public class MockGreetingProfile implements QuarkusTestProfile {
     public String getConfigProfile() { <3>
         return "test";
     }
+
+    @Override
+    public List<TestResourceEntry> testResources() { <4>
```
Comment (gastaldi) [inline docs/src/main/asciidoc/getting-started-testing.adoc:510]: I think you're missing an import to `java.util.List`?

Comment (geoand) [reply docs/src/main/asciidoc/getting-started-testing.adoc:511]: Fixed

**Pull request merged**